File: android/src/main/kotlin/com/usercentrics/sdk/flutter/serializer/ConsentDisclosureSerializer.kt
package com.usercentrics.sdk.flutter.serializer

import com.usercentrics.sdk.v2.settings.data.ConsentDisclosure
import com.usercentrics.sdk.v2.settings.data.ConsentDisclosureObject
import com.usercentrics.sdk.v2.settings.data.ConsentDisclosureSDK

internal fun ConsentDisclosureObject?.serializer(): Any? {
    if (this == null) {
        return null
    }
    return mapOf(
        "disclosures" to disclosures.map { it.serializer() },
        "sdks" to sdks.map { it.serializer() }
    )
}

private fun ConsentDisclosureSDK.serializer(): Map<String, Any> {
    return mapOf(
        "name" to name,
        "use" to use
    )
}

internal fun ConsentDisclosure.serializer(): Any {
    return mapOf(
        "identifier" to identifier,
        "type" to (type?.name ?: ""),
        "name" to name,
        "maxAgeSeconds" to maxAgeSeconds,
        "cookieRefresh" to cookieRefresh,
        "purposes" to purposes,
        "domain" to domain,
        "description" to description,
    )
}

File: ios/Classes/Serializer/ConsentDisclosureSerializer.swift
import Usercentrics

extension ConsentDisclosureObject {
    func serialize() -> Any {
        return [
            "disclosures": self.disclosures.map { $0.serialize() },
            "sdks": self.sdks.map { ["name": $0.name, "use": $0.use] }
        ] as [String: Any]
    }
}

extension ConsentDisclosure {
    func serialize() -> Any {
        return [
            "identifier": self.identifier as Any,
            "type": self.type?.name ?? "",
            "name": self.name as Any,
            "maxAgeSeconds": self.maxAgeSeconds as Any,
            "cookieRefresh": self.cookieRefresh,
            "purposes": self.purposes,
            "domain": self.domain ?? "",
            "description": self.description,
        ] as [String : Any]
    }
}
